pymaker/gas.py

```diff
@@ -137,3 +139,45 @@ def get_gas_price(self, time_elapsed: int) -> Optional[int]:
             result = min(result, self.max_price)
 
         return result
+
+
+class GeometricGasPrice(GasPrice):
+    """Geometrically increasing gas price.
+
+    Start with `initial_price`, then increase it every 'every_secs' seconds by a fixed coefficient.
+    Coefficient defaults to 1.125 (12.5%), the minimum increase for Parity to replace a transaction.
+    Coefficient can be adjusted, and there is an optional upper limit.
+
+    Attributes:
+        initial_price: The initial gas price in Wei i.e. the price the transaction is originally sent with.
+        every_secs: Gas price increase interval (in seconds).
+        coefficient: Gas price multiplier, defaults to 1.125.
+        max_price: Optional upper limit, defaults to None.
+    """
+    def __init__(self, initial_price: int, every_secs: int, coefficient=1.125, max_price: Optional[int] = None):
+        assert (isinstance(initial_price, int))
+        assert (isinstance(every_secs, int))
+        assert (isinstance(max_price, int) or max_price is None)
+        assert (initial_price > 0)
+        assert (every_secs > 0)
+        assert (coefficient > 1)
+        if max_price is not None:
+            assert(max_price > 0)
+
+        self.initial_price = initial_price
+        self.every_secs = every_secs
+        self.coefficient = coefficient
+        self.max_price = max_price
+
+    def get_gas_price(self, time_elapsed: int) -> Optional[int]:
+        assert(isinstance(time_elapsed, int))
+
+        if time_elapsed < self.every_secs:
+            return self.initial_price
+        result = self.initial_price
+        for second in range(math.floor(time_elapsed/self.every_secs)):
+            result *= self.coefficient
```

Comment:
Should mention: I didn't want a `for` loop here, but raising the coefficient to the power of the elapsed seconds produces  `OverflowError` for values on the order of 10000 seconds (and up).